lighthouse-core/gather/gatherers/devtools-log.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This gatherer collects all network and page devtools protocol traffic during the timespan/navigation.
+ * This protocol log can be used to recreate the network records using lib/network-recorder.js.
+ */
+
+const MessageLog = require('../devtools-log.js');
+const FRGatherer = require('../../fraggle-rock/gather/base-gatherer.js');
+
+class DevtoolsLog extends FRGatherer {
+  static symbol = Symbol('DevtoolsLog');
+
+  /** @type {LH.Gatherer.GathererMeta} */
+  meta = {
+    symbol: DevtoolsLog.symbol,
+    supportedModes: ['timespan', 'navigation'],
+  };
+
+  _messageLog = new MessageLog(/^(Page|Network)\./);
```

Comment:
nit: Does it make a difference if `_messageLog` is defined in the constructor? It seems weird to be defining the private members in two different places.

Reply:
no difference really, I was following the driver pattern that started with anything that could use this syntax did and everything else was in the constructor.

I can move it to the constructor though